Automatically locate Sphnix conf.py by searching the workspace and allow the user to pick if there are more than one such file

TormodLandet opened the pull request:
- [x] Find all instances of `conf.py` in the current workspace
- [x] Find all `conf.py` files in parent directories (if a ReST file is opened directly and not as a part of a workspace, fixes issue #91).
- [x] Let the user select which Sphinx directory to use if multiple such directories are found (fixes issue #10)
- [x] Remember the users choice of selected conf.py file
- [x] Fallback to plain rst2html if no Sphinx config is found (could replace pr #61 and pr #42) 
- [x] Show the current active conf.py file in the status bar

TormodLandet commented:
The pull request is updated to find all conf.py files in the workspace and let the user decide which one to use from a drop down (based on the C# extension code you pointed out).

The choice should be cached somehow. Either per rst file or per preview window (or something else?). What do you think?

lextm commented:
Thanks. Will start reviewing.

lextm commented:
The latest change is a big step forward, but its user experience does not look good enough,

1. It seems that every preview operation requires a selection of `conf.py` file.
1. There is no place showing which `conf.py` file is selected currently. The C# extension adds a section in status bar to show that.
1. The selection dialog currently shows a list of directories. Personally I think showing a list of `conf.py` files should be more intuitive.

TormodLandet commented:
I think I have addressed all your comments in the updated pull request.

Also, since the drop-down menu allows some flexiblity I added the option to fallback to the plain docutils rst2html.py command if no Sphinx config is found. Sphinx depends on docutils which provides the rst2html.py command, so it should work for most people automatically (I added a configuration option to set the path if it does not work).

TormodLandet commented:
Anything you need from me on this? I can fix the new merge conflict if you like, but I will probably not take the time if you do not have time to merge this now anyway. This PR has been done from my side since Jul 1, unless I have missed some of your comments?

lextm commented:
Thanks. I am too busy to finish reviewing, but should be able to work on it some time by the end of this month. Appreciate all your help.

lextm commented:
Overall it looks perfect, but one thing is not intuitive.

It seems that I need to click the new widget in status bar to clear the cached `conf.py`, and then the next preview will prompt and users can make a new selection.

It is very hard to discover this. Not sure if we can trigger a prompt immediately when the status bar widget is clicked.

lextm commented:
You might have to rebase, as I made some changes and triggered some conflicts.

TormodLandet commented:
Thanks! I will see if I can find time this weekend to rebase and look into the cache clearing issue

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vscode-restructuredtext/vscode-restructuredtext?pullRequest=92) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/vscode-restructuredtext/vscode-restructuredtext?pullRequest=92) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: TormodLandet<br/>:x: lextm<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/vscode-restructuredtext/vscode-restructuredtext?pullRequest=92) it.</sub>

TormodLandet commented:
Seems like we found time to work on this at the same time :-) Do you have the merge with master done or do you want me to push my merge?

lextm commented:
I reverted some changes on master. So merge it now. Thanks for all the help.

lextm commented:
@TormodLandet I think there is still a minor part missing, which if you like you can work on another pull request.

We should save the current select config somewhere permanent (.vscode\settings.json I assume), so that after closing and opening VSCode, the previous selection remains and users don't need to select again.

Let me know if you rather let me work on it :)

TormodLandet commented:
I will let you know if I start coding on it. I guess the major part (for me) is to find a good example of how other plugins handle this type of project specific caching. When a good example is found it should be relatively easy to adapt. It is already a bit late here, so I do not have time today at least

rizickus commented:
Is this feature already released? I can not find any release notes in this project. Using 78.0.0 fails to get `conf.py` file in `docs` folder and does not provide any preview options except rst2html.

lextm commented:
@rizickus Please open a new issue with more details.
